images/hook/entrypoint.sh

```diff
@@ -21,9 +20,11 @@ if [ $1 = "update" ]; then
         rig delete deployments/heapster --resource-namespace=$namespace --force
 
         echo "---> Deleting old 'influxdb' resources"
-        # Get node name where influxdb pod scheduled to patch deployment
-        # and reschedule the pod on the same node after update
-        NODE_NAME=$(kubectl --namespace=kube-system get pod -l app=monitoring,component=influxdb -o jsonpath='{.items[0].spec.nodeName}')
+        # Patch influxdb deployment but keep the pod scheduled to the same node after the update
+        if kubectl --namespace=$namespace get deployment influxdb --ignore-not-found=false 2>/dev/null; then
+            NODE_NAME=$(kubectl --namespace=$namespace get pod -l app=monitoring,component=influxdb -o go-template --template='{{(index .items 0).spec.nodeName}}')
```

Comment:
One thing to note is that `NODE_NAME` is being initialized conditionally but used unconditionally below - do you need to make sure it's set when you create the patch for influxdb deployment?

Reply:
Good catch! Will add check for variable emptyness.